zip-0155.rst

```diff
@@ -0,0 +1,246 @@
+
+
+Specification
+=============
+
+The ``addrv2`` message is defined as a message where ``pchCommand == "addrv2"``.
+It is  serialized in the standard encoding for P2P messages. Its format is similar
+to the  current ``addr`` message format described in [#Bitcoin-addr-message]_, with
+the difference that the fixed 16-byte IP address is replaced by a network ID and a
+variable-length address, and the services format has been changed to
+[#Bitcoin-CompactSize]_.
+
+This means that the message contains a serialized vector of the following structure:
+
+  +--------------+-----------------+---------------------+----------------------------------------------------------------+
+  | Bytes        | Name            | Data Type           | Description                                                    |
+  +--------------+-----------------+---------------------+----------------------------------------------------------------+
+  | 4            | ``time``        | ``uint32``          | Time that this node was last seen as connected to the network. |
+  |              |                 |                     | A time in Unix epoch time format.                              |
+  +--------------+-----------------+---------------------+----------------------------------------------------------------+
+  | ``varies``   | ``services``    | ``CompactSize``     | Service bits. A CompactSize-encoded bit field that is 64 bits  |
+  |              |                 |                     | wide.                                                          |
+  +--------------+-----------------+---------------------+----------------------------------------------------------------+
+  | 1            | ``networkID``   | ``uint8``           | Network identifier. An 8-bit value that specifies which        |
+  |              |                 |                     | network is addressed.                                          |
+  +--------------+-----------------+---------------------+----------------------------------------------------------------+
+  | ``varies``   | ``sizeAddr``    | ``CompactSize``     | The length in bytes of ``addr``.                               |
+  +--------------+-----------------+---------------------+----------------------------------------------------------------+
+  | ``sizeAddr`` | ``addr``        | ``uint8[sizeAddr]`` | Network address. The interpretation depends on networkID.      |
+  +--------------+-----------------+---------------------+----------------------------------------------------------------+
+  | 2            | ``port``        | ``uint16``          | Network port. If not relevant for the network this MUST be 0.  |
+  +--------------+-----------------+---------------------+----------------------------------------------------------------+
+
+One message can contain up to 1,000 addresses. Clients MUST reject messages with
+more addresses.
+
+Field ``addr`` has a variable length, with a maximum of 512 bytes (4096 bits). Clients
```

Comment:
Would it be worthwhile to explicitly rename `addr` to `addrv2`?

Reply:
That's the name of the message. I think `addr` is fine for the field (as in BIP 155).